src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/backfill/GetAlertsForCaseNotes.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.hmppsalertsapi.backfill
+
+import org.springframework.data.jpa.domain.Specification
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.Alert
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.AlertCode
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.AlertType
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.AuditEvent
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Service
+class GetAlertsForCaseNotes(private val alertRepository: AlertRepository) {
+  fun get(prisonNumber: String, from: LocalDate, to: LocalDate): List<CaseNoteAlert> =
+    alertRepository.findAll(caseNoteSpecification(prisonNumber, from, to)).map { it.forCaseNotes() }
+
+  private fun caseNoteSpecification(prisonNumber: String, from: LocalDate, to: LocalDate) =
+    Specification<Alert> { alert, _, cb ->
+      val pn = cb.equal(alert.get<String>("prisonNumber"), prisonNumber)
+      val created = cb.between(alert.get("createdAt"), from, to)
```

Comment:
Is it best to go through the alert audit event actionedAt value? All created events in the date range would be in scope plus any updates where activeFrom/ToUpdated = true